packages/cerebral/src/Module.js

```diff
@@ -1,35 +1,24 @@
 class Module {
-  constructor (path, module, controller) {
-    this.name = path.slice().pop()
-    this.path = path
-    this.controller = controller
-    this.modules = {}
+  constructor (controller, path, moduleDescription) {
+    const moduleStub = {
+      controller,
+      path: path.join('.'),
+      name: path.slice().pop()
+    }
 
-    const moduleDescription = typeof module === 'function' ? module(this) : module
+    const module = typeof moduleDescription === 'function' ? moduleDescription(moduleStub) : moduleDescription
```

Comment:
Yes, changes was needed. Because there was creating new object instance by copy some properties from description object: https://github.com/cerebral/cerebral/pull/595/files#diff-da938b16e674eaf4bf8d7cd785d662caL15.

In fact main trick here is return existing object from constructor.